### PR TITLE
Docs/419377  response schema api endpoints

### PR DESCRIPTION
**Pull request by whitewaterdesign**

### CODE REVIEW

File: app/routes/cdo.js [hunk 1]
```diff
@@ -52,7 +55,16 @@ module.exports = [
   {
     method: 'GET',
     path: '/cdo/{indexNumber}',
-    options: { tags: ['api'] },
+    options: {
+      tags: ['api'],
+      notes: ['Returns the full CDO with dog, person and exemption details'],
```
Comment (jbarnsley10) [inline app/routes/cdo.js:60]: I like the addition of 'notes' on these methods

File: app/routes/dog.js [hunk 2]
```diff
@@ -31,14 +41,26 @@ module.exports = [
...
+      }
+    },
     handler: async (request, h) => {
       const indexNumber = request.params.indexNumber
       let ownerDao
       let owner
 
       try {
-        if (request.query.includeDogs === 'true') {
+        if (request.query.includeDogs) {
```
Comment (jbarnsley10) [inline app/routes/dog.js:63]: Have we got anywhere that might make a call using ...&includeDogs=false ?

File: app/schema/dogs/get.js [hunk 3]
```diff
@@ -8,6 +8,40 @@ const dogsQueryParamsSchema = Joi.object({
...
+    id: Joi.number(),
+    indexNumber: Joi.string(),
+    name: Joi.string().allow(null).allow(''),
+    breed: Joi.string(),
+    colour: Joi.string().allow(null).allow(''),
+    sex: Joi.string().allow(null).allow(''),
+    dateOfBirth: Joi.date().allow(null),
+    dateOfDeath: Joi.date().allow(null),
+    tattoo: Joi.string().allow(null).allow(''),
+    microchipNumber: Joi.date().allow(null),
```
Comment (jbarnsley10) [inline app/schema/dogs/get.js:22]: microchips will be string, not date

File: app/schema/person/dog-owner.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  dogReference: Joi.string(),
+  id: Joi.number(),
+  indexNumber: Joi.string(),
+  microchipNumber: Joi.string().allow('').allow(null),
+  microchipNumber2: Joi.string().allow('').allow(null),
+  name: Joi.string().allow('').allow(null),
+  sex: Joi.string().allow('').allow(null),
+  status: Joi.string(),
+  tattoo: Joi.string().allow('').allow(null),
+  birthDate: Joi.string().allow('').allow(null)
```
Comment (jbarnsley10) [inline app/schema/person/dog-owner.js:15]: birthDate will be a date

**Review by jbarnsley10 — CHANGES_REQUESTED** [review]
A few types are wrong but otherwise all great

File: app/routes/cdo.js (hunk 1 above)
Comment (whitewaterdesign) [reply app/routes/cdo.js:60]: this comes out in swagger docs as well

File: app/schema/dogs/get.js (hunk 3 above)
Comment (whitewaterdesign) [reply app/schema/dogs/get.js:22]: thanks, good catch

File: app/schema/person/dog-owner.js (hunk 4 above)
Comment (whitewaterdesign) [reply app/schema/person/dog-owner.js:15]: yeah, was having issues thinking it was to do with this, seems it wasn't - changed back to date now

**Review by jbarnsley10 — APPROVED** [review]
Great

**Pull request merged**